blocks/bricks/sequence_generators.py

```diff
@@ -436,6 +436,9 @@ def emit(self, readouts):
 
     @application
     def cost(self, readouts, outputs):
+        # WARNING: unfortunately this application method works
+        # just fine when `readouts` and `outputs` have
+        # different dimensions. Be careful!
         probs = self.probs(readouts)
```

Comment:
Asserts are really nice, but they break optimization. One of Theano guys told me, that they have an idea to delete them from the graph in `fast_run`, something like that. 

I like your idea @bartvm! Can you create an issue so that we do not forget about it?